lib/components/widgets/wrapper/code_display.dart

```diff
@@ -0,0 +1,141 @@
+  /// [keywordTextStyle], [stringTextStyle], and [punctuationTextStyle]
+  /// parameters can be used to customize
+  /// the styles of different parts of the code.
+  ///
+  /// The [padding] parameter specifies the padding of the code container.
+  CodeDisplay(
+    this.code, {
+    this.codeColorTheme,
+    this.width,
+    this.height,
+    this.boxDecoration,
+    this.commentTextStyle,
+    this.baseTextStyle,
+    this.classTextStyle,
+    this.keywordTextStyle,
+    this.stringTextStyle,
+    this.punctuationTextStyle,
+    this.padding,
+    super.key,
+  }) : formatedCode = formatCode(code);
+
+  /// The code to be displayed.
+  final String code;
+
+  /// The formatted code.
+  final String formatedCode;
+
+  /// The height of the code display.
+  final double? height;
+
+  /// The width of the code display.
+  final double? width;
+
+  /// The color theme for syntax highlighting.
+  final CodeDisplayColorThemes? codeColorTheme;
+
+  /// The box decoration of the code display.
+  final BoxDecoration? boxDecoration;
+
+  /// The text style for comments.
+  final TextStyle? commentTextStyle;
+
+  /// The base text style.
+  final TextStyle? baseTextStyle;
+
+  /// The text style for classes.
+  final TextStyle? classTextStyle;
+
+  /// The text style for keywords.
+  final TextStyle? keywordTextStyle;
+
+  /// The text style for strings.
+  final TextStyle? stringTextStyle;
+
+  /// The text style for punctuation.
+  final TextStyle? punctuationTextStyle;
+
+  /// The padding of the container.
+  final EdgeInsets? padding;
+
+  /// Returns a [TextStyle] with the specified [color]
+  /// if the [textStyle] is not null,
+  /// otherwise returns a [TextStyle] with the specified [color],
+  /// or an empty [TextStyle].
+  TextStyle getTextStyle(TextStyle? textStyle, Color? color) {
+    if (textStyle != null && textStyle.color != null) {
+      return textStyle;
+    }
+
+    return (textStyle ?? const TextStyle()).copyWith(color: color);
+  }
+
+  /// Returns a [BoxDecoration] with the specified [color]
+  /// if the [boxDecoration] is not null,
+  /// otherwise returns a [BoxDecoration] with the specified [color],
+  /// or an empty [BoxDecoration].
+  BoxDecoration getBoxDecoration(BoxDecoration? boxDecoration, Color? color) {
+    if (boxDecoration != null) {
+      return boxDecoration.color != null
+          ? boxDecoration
+          : boxDecoration.copyWith(color: color);
+    }
+
+    if (color != null) {
+      return BoxDecoration(color: color);
+    }
+
+    return const BoxDecoration();
+  }
+
+  @override
+  Widget build(BuildContext context) => Container(
+        key: const Key('CodeDisplayContainer'),
+        padding: padding,
+        decoration: getBoxDecoration(
+          boxDecoration,
+          codeColorTheme?.backgroundColor,
+        ),
+        child: DartCodeViewer(
+          formatedCode,
+          width: width,
+          height: height,
+          showCopyButton: false,
```

Comment:
This also could be an option to users to customise by setting the default to `false`. That way, if they want to show slides + code demo, they could just copy it from here if needed.

Not sure if there would be any limitations.

Reply:
Ok, now that you mention it, I will implement it. The issue here was it only works with MaterialApp since the copy button triggers a ScaffoldMessenger to display a "copied" message. I don't think we need that, so I will update `dart_code_viewer_x` so we can decide wether it should trigger or not.